### ISSUE TITLE
Support M2M numbers

### ISSUE THREAD

**Issue by ameenross** [googlei18n/libphonenumber#680]
M2M (machine-to-machine) numbers are currently not supported by libphonenumber. There have been requests to support such numbers from various countries:
- #332 
- #550 
- #657

I can only speak for the Netherlands, as I'm a Dutch citizen. The 097-range is a number range designated as M2M, and intended for consumer use (tablet/laptop mobile internet). Explosive growth is expected. I'd think that similar trends can be seen in other countries, so critical mass will be reached soon enough. Therefore I think it's important for libphonenumber to support this.

**Comment by hagbard** [comment]
We have no plans to support this at the moment, but might in the future. The phonenumber library is currently aimed at cases where numbers can be dialed to reach a "human" endpoint.
We will have to look carefully at whether it's ok to at M2M numbers without breaking existing users (if we say it's a "valid" number, have people been assuming that means it's going to reach a human?).

**Comment by ameenross** [comment]
Like I said, my wife could receive calls. Only some carriers would reject the dialing of her number, which apparently has nothing to do with my wife's connection, but is a restriction on the other end.

**Comment by ameenross** [comment]
As for the practical problem of people using libphonenumber to find out whether a number is a valid "regular" (whatever that means) number, you could perhaps add a new method of validation. Update the old method to say that M2M is excluded for the reason you mentioned, and point to the new method for people with lower requirements (such as "SMSable" numbers).

**Comment by TMcManus** [comment]
We would like to see M2M numbers be supported by libphonenumber as well. Implementing a new method in addition to the validity check would be acceptable. It should also be pointed out that M2M range numbers may or may not be voice enabled depending on the country.

**Comment by padmaksha** [comment]
Hi

As @hagbard said, this requires a bit more thought. Will update this thread with new information once we have it. Thanks for the reports, it is important for us to get your feedback.

Thanks and Regards
Padmaksha

**Comment by andyst** [comment]
We're actually working towards a solution for M2M and as a first step, we're communicating consistently.

 - We've created a FAQ and M2M is one of the first FAQ entries. https://github.com/googlei18n/libphonenumber/blob/master/FAQ.md#what-about-m2m-machine-to-machine-numbers
 - The FAQ entry explains why we don't understand the issue yet and how to engage with us.

Let's use this issue here to make progress on the open issues.

**Comment by ameenross** [comment]
> to date we only accept ranges that a human might call or send an SMS to. M2M numbers would violate this assumption

As I indicated above, my wife can be called on her M2M number just fine, except some carriers block or "don't understand it" for some reason. SMS works just fine as well.

The Dutch (and some other countries') new M2M ranges are meant for mobile internet devices (a lot of them tablets). As such, they will be widely used by the consumer market and not really "machines" at all.

> We don't think the lack of M2M support hinders any of those use-case.

@daniele-athome (developer of Kontalk) has [indicated almost a year ago](https://github.com/googlei18n/libphonenumber/issues/332#issuecomment-96980066) that Kontalk uses libphonenumber to validate phone numbers people try to register with. I have also indicated a long time ago that people buying tablets with mobile internet in the Netherlands will get an M2M number (and have been getting since 2014), and therefore will not be able to use Kontalk (or Whatsapp for that matter).

So basically, all questions that are posed in that FAQ have really already been answered at least once. If there are any additional questions, let's hear it.

**Comment by Gastomagic** [comment]
Hi all

I'm head of M2M Operations at JT Global.
I have been on other threads attempting to get our 15 digit non-geographic +88236 range added.

 Can anyone tell me what I need to get the range added to the library? What information is required?

the biggest problem I have is that Twilio use the library - and I have many clients that use the Twilio service for sending shoulder taps and configuration messages via SMS to their SIMs / devices. There are countless applications for M2M of course.

I've suggested that Twilio make a manual addition to their own numbering scheme but they're adamant they follow the library - despite it being incomplete.

This essentially leads to me having to reprovision services with +447 numbers (which are in extremely short supply).

Any wisdom appreciated

**Comment by andyst** [comment]
I've started collecting information about the all the different requests and M2M in general. Here's an update so far:

The missing standardization & inconsistent handling of M2M across countries is worrisome. For libphonenumber, we need to be able to classify numbers into fixedLine, mobile and other categories. Adding a new type can't be done lightly and so far it doesn't look like it makes sense to add M2M as a type due to the varying ways it's used in different countries. Thus we'd probably want to add M2M ranges under the "mobile" section where that applies.

It's going to be imperative that there's clear information for each M2M range that should be added. For any given country or even carrier, I'm very concerned that it may be a lot of effort to get the needed classification (readability, mobile/fixedLine/..., price, use-cases) which may make it prohibitively expensive to be maintained. The information in the ITU docs for M2M are too sparse for now and we need to get such information from an official document (ITU / government / carrier). We may end up with a model where the subset of M2M ranges may be added for which the requester provides all the information with official docs supporting the claimed classifications.

We should probably start by collecting this information for each requested M2M range:
*   Country
*   Range
*   Prefix/Length
*   Pricing compared to existing types we support (fixed-line, mobile, shared costs, premium rate)
*   Reachability from fixed-line/mobile carriers
*   Reachability from overseas
*   Reachability for voice and SMS
*   Whether the above information applies to the complete range or depends on other factors.
*   Usecase
*   Official documentation serving as evidence for the above information

**Comment by bdewater** [comment]
- Country: The Netherlands
- Prefix/Length: 097x followed by 8 digits. Currently valid integers for x are 0 (public numbers) and 9 (network internal usage). 1-8 are reserved for future use if the need arises.
- Reachability: the device must be able to receive SMS messages [(source)](http://tweakers.net/nieuws/96131/providers-mogen-tablets-geen-06-nummer-meer-geven.html)
- Pricing should be the same as 'normal' mobile numbers.
- Usecase: 'electronic communication services between automated applications' and 'other electronic communication services', where the userfriendlyness and recognizability of the number is not a priority. The announcement mentions alternatives like IP, but acknowledges that not all applications can or need to be connected to the internet at all times (e.g. to save cost).
- Official documentation serving as evidence for the above information: [Official announcement in the Staatscourant](https://www.acm.nl/nl/publicaties/publicatie/10289/Besluit-nummers-voor-elektronische-communicatiediensten-voor-geautomatiseerde-toepassingen/), [Telecommunications law numbering plan](http://wetten.overheid.nl/BWBR0010198/)

A bit of back story: the pool of 'normal' 06 mobile phone numbers was depleting faster than anticipated due to the popularity of tablets, smart energy meters and other devices that do not use voice as a primary means of communication. In order to avoid having to switch from 10 to 11 digit long numbers, the ACM (called OPTA back then) announced that the 097 range is created for these devices. As of June 1st, 2014 [it is mandatory](http://tweakers.net/nieuws/96131/providers-mogen-tablets-geen-06-nummer-meer-geven.html) for Dutch providers to issue 097 numbers for new subscribers when they are using it for M2M services. Provider Tele2 received [a fine](https://www.acm.nl/nl/publicaties/publicatie/13212/Besluit-op-bezwaar-last-onder-dwangsom-Tele2/) for not complying with the law.

FWIW I've contributed support for M2M numbers to this Ruby library: https://github.com/floere/phony/commit/38982e7d5d9d1a4f487362a54cf9344ad71ee544

**Comment by Gastomagic** [comment]
Hi Andy

Thanks very much for the attention on this – much appreciated:


  *   Country – Non-geographic range – just assigned by the ITU to JT Global (Formally Jersey Telecom)
  *   Range - +882360000000000 to +882369999999999
  *   Prefix/Length +88236XXXXXXXXXX (populated to 15 digits)
  *   Pricing compared to existing types we support (fixed-line, mobile, shared costs, premium rate) Terminating price is identical to our normal mobile ranges (+447797, +447509, +447937)
  *   Reachability from fixed-line/mobile carriers  - Currently open to SMS only but work is ongoing to make the full range voice routable also
  *   Reachability from overseas - Currently open to SMS only but work is ongoing to make the full range voice routable also
  *   Reachability for voice and SMS - Currently open to SMS only but work is ongoing to make the full range voice routable also
  *   Whether the above information applies to the complete range or depends on other factors. – Full range, although it depends on the mobile operator having added the range to their routing. The vast majority of our 550 roaming partners have already added the range (the routing information was released over 2 years ago via the IR21 document). In cases where we find it is not properly routed we contact the operator immediately and get it remedied.
  *   Usecase – All manner of M2M use cases – from lone worker / alarm systems, to basic IOT devices
  *   Official documentation serving as evidence for the above information – Andy can you tell me what information you need here? How about a copy of our IR21 document?
Thanks again
TG


Toby Gasston  |  Design & Integration Lead M2M & SR |  Wholesale Business Unit
DDI: +44 1534 882887   |  Mobile: +44 7797 857 279   |  Fax: +44 1534 882434   |  email: toby.gasston@jtglobal.com<mailto:toby.gasston@jtglobal.com>
JT, PO Box 53, No 1 The Forum, Grenville Street, St Helier, Jersey, JE4 8PB
JT, 24 High Street, St Peter Port, Guernsey, GY1 2JU, Registered Company no. 39971
www.jtm2m.com<http://www.jtm2m.com/>
www.jtglobal.com<http://www.jtglobal.com/>

[Description: JT]

**Comment by ameenross** [comment]
Is there an update on this?

**Comment by ameenross** [comment]
I guess we must conclude that Google has no interest of implementing this currently. The community may be better off finding a different solution.

**Comment by Gastomagic** [comment]
I agree Ameen

I am genuinely surprised at the unnecessary bureaucracy around such a simple concept from a company so integral to communication.

“Hi Google – can you please add this range to your numbering scheme?”
“hi – yes of course – no problem”

Seems pretty simple to me but I’ve not been chasing it for more than 6 months

**Comment by andyst** [comment]
Thanks for bumping the thread and thanks for the clarifications in March.
We understand your frustration as there wasn’t an update from our side in a while.

So far we have received more details about two M2M ranges in this thread (NL 097 and international +88236).

@Gastomagic: Yes, please send us your IR.21 doc. See https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTING.md#provide-supporting-evidence on how to send it to us.

What M2M phone numbers are and how they are used is still rapidly developing and not much standardized worldwide. That’s why we are surveying here the capabilities / properties of the M2M ranges you’d like to see supported by libphonenumber.

While we understand the concrete use cases for M2M phone numbers that you’ve mentioned (thank you!), the challenge is to decide how to best add support for these ranges. Whether it’s under an existing category or a new (e.g. `M2M` category) or another alternative.

Introducing a new category (such as `FIXED_LINE`, `MOBILE`, `VOIP`) in libphonenumber can’t be done lightly. This is controversial for M2M as there are few aspects that all the M2M ranges have in common, especially regarding voice & SMS capability.

## Why not as `MOBILE`
Initially our plan was to support select M2M ranges as `MOBILE`. Given that M2M numbers are still emerging in many countries and that they’re not all defined the same way, this seemed like a pragmatic approach. But it turned out that this won’t work. libphonenumber users have come to expect that `MOBILE` numbers generally have both voice and SMS capabilities. So far these criteria aren’t met by the M2M ranges that we’re aware of.

## A new `M2M` category?
We’re currently evaluating if there’s a strong case to add a `M2M` category. We see some parallels to `VOIP` (yet at least with voip users have an expectation of voice capability). Maybe there’s a case for adding a new category if we have official evidence for M2M ranges in 5+ countries and if all that libphonenumber users need here is validation (considering such numbers as valid) and basic formatting / canonicalization (to E.164) support. We need a critical mass of verified countries / ranges to move forward with this approach.

## Help us
*   @Gastomagic: See above, please share your IR.21 with us.
*   **Please confirm** that all that you expect from libphonenumber is the following: parse successfully, isValid should return true, format in E.164 format-type. libphonenumber wouldn’t classify these numbers as `MOBILE`.
*   In terms of details & evidence, we still need more information for some countries / ranges.
    *   We’re good for **NL** 0970.
    *   For **international +88236** we need the IR.21 or an official document by the government.
    *   For **South Africa** +27872405086xxxxx the latest ITU doc seems to be out of date (classifies 087 as VOIP of shorter length).
    *   For **Norway** the range is confirmed by gov. docs as M2M but it’d be great to learn more about how M2M is officially intended to be used there in terms of cost, voice, SMS.
    *   Same for **Sweden**.
    *   **Belgium** 77\d{11}: Need to carefully read through the official doc linked from #550 to figure out more details about voice, SMS & cost and other relevant aspects.

We can hardly justify the additional complexity of a new category if it’s just for a couple countries. Please help us to build the case with strong and clear evidence about the requirements and capabilities.

Thanks!
Andy on behalf of the team

**Comment by Gastomagic** [comment]
Thanks Andy

I have signed the CLA and sent the document over

TG

**Comment by ameenross** [comment]
First of all: thanks for the update. Waiting can be quite frustrating, especially when you're building software that blocks on an external dependency.

> MOBILE numbers generally have both voice and SMS capabilities. So far these criteria aren’t met by the M2M ranges that we’re aware of.

I have repeated myself numerous times now that the Dutch M2M range *does* have (non-VOIP) voice and SMS capabilities. @bdewater also pointed this out, and IIRC I've seen at least another mention for other ranges. At least for these ranges, I really don't see any reason for a new category.

Maybe there are ranges that do deserve a new category. If that is the case then I think the solution would be *both*. Some ranges fall under the old mobile category, whereas the others will fall under a new category or perhaps another old category.

**Comment by andyst** [comment]
@ameenross: It's not sufficient that some M2M SIM card users in NL use the number with a device that has SMS and voice capability. NL requires that devices without SMS capability get a M2M rather than a mobile number. The official documentation describes various use cases of M2M but SMS and voice are the exception, not the rule.
We can only add ranges to MOBILE which generally are intended for SMS & voice. If only a fraction of the users / numbers of such a range can receive/send SMS and voice, then we're doing wrong by existing libphonenumber clients if we declared it as MOBILE as existing clients depend on these capabilities.

**Comment by ameenross** [comment]
Alright I get your point. I would contend, though, that tablets have voice and SMS capability, and tablet users are not just a tiny fraction of the M2M users. I should also point out that the mobile internet laptop dongle I've used in the past supported SMS in some form.

If the number categories are based on *use-case* however, that might be problematic. Dutch (at least) M2M numbers have wildly varying use-cases. For example: smart meters and mobile (tablet/laptop) internet are completely different beasts.

**Comment by andyst** [comment]
@Gastomagic I don't see a new email in the libphonenumber-supporting-docs mailbox. Maybe the email bounced because you didn't join the mailing list yet? You can send us your email address to be whitelisted by the contacting the owner, see https://groups.google.com/forum/#!managemembers/libphonenumber-supporting-docs .

**Comment by Gastomagic** [comment]
Thanks Andy

I went through the process of filling out the required documentation, then creating a group in order to submit it etc etc – they had the application voided with a blank reason field.

Anyway – I’ve mailed the owner now so will see how we get on

TG

**Comment by bdewater** [comment]
> Introducing a new category (such as FIXED_LINE, MOBILE, VOIP) in libphonenumber can’t be done lightly. This is controversial for M2M as there are few aspects that all the M2M ranges have in common, especially regarding voice & SMS capability.

If the library strives to give such information to its users, it would make sense to define these capabilities per country/range instead of on the entire category for M2M.

> Belgium: Need to carefully read through the official doc linked from #550 to figure out more details about voice, SMS & cost and other relevant aspects.

The linked document in that issue is an addendum to the original decision at http://www.bipt.be/en/operators/telecommunication/Numbering/regulation/decision-of-the-bipt-council-of-6-september-2011-on-the-determination-of-the-numbering-plan-for-m2m-communication

**Comment by bdewater** [comment]
According to the [ACM's 2015 number availability report](https://www.acm.nl/nl/publicaties/publicatie/16022/Schaarsterapportage-2015/) 11.8 million (up from 8.4 million in 2014) Dutch M2M 0970 numbers haven been issued. There's still a shortage of 06 numbers with 90.8% (up from 90.7%) of the available mobile 06 numbers issued.

The ACM also published a [fact sheet for providers](https://www.acm.nl/nl/publicaties/publicatie/16001/Informatie-over-097--en-06-nummers--factsheet-voor-telecomaanbieders/). Some interesting bits:

- The fact sheet emphasizes the recognizability and user friendlyness of the number over capability ("numbers are usually entered only once and then dialled automatically")
- Using 06 numbers sms-only services (like tablets) is prohibited
- It confirms that 0970 numbers can be used for voice (M2H alarm systems, H2H automatic dialing emergency services after a car crash)
- Providers must have a clause in their contracts that prohibits M2M usage of 06 numbers by their customers
- Providers must proactively ensure that 06 numbers aren't used for M2M after issuing, the ACM can revoke the number or fine the provider if they knowingly did not take action

Given this there will be more tablet users every year that will run in to trouble because applications that use libphonenumber will refuse to send an SMS to their device.

**Comment by adebree** [comment]
A NL / Dutch use case I've come in contact with using the 097 range is in the emergency voice systems in elevators. Those can both be used to dial from to a emergency callcenter (automatic dialing from an embedded system hosting the mobile SIM having the 097 number) as well as being used to calling back for voice purposes. Support from libphonenumber for this range would therefore be greatly appreciated.

**Comment by Nazta** [comment]
https://github.com/googlei18n/libphonenumber/blob/master/resources/ShortNumberMetadata.xml#L8440

5 digits number are blocked.

reference/source: http://eng.nkom.no/market/numbering/e.164-numbering-plan/5-digit-numbers-02000-09999

**Comment by lararennie** [comment]
What is this comment about? Do you want to file an issue with us about short numbers in Norway? Or something else?

**Comment by Nazta** [comment]
Well yes, searched for m2m/shortcode open issues, and this was the first one. Should probably have open a new issue I guess.

**Comment by lararennie** [comment]
M2M and short codes are *not* considered the same thing. Please file a new issue and see our contribution guidelines for the template to use. Thanks!

https://github.com/googlei18n/libphonenumber/blob/master/CONTRIBUTING.md

**Comment by bdewater** [comment]
> Maybe there’s a case for adding a new category if we have official evidence for M2M ranges in 5+ countries

So far I'm seeing the Netherlands, South Africa, Norway, Sweden, Belgium, France and a non-geographic range mentioned in or linked to this issue. Is there a possibility that we can move forward with some kind of minimal use case that starts recognizing these real in-use numbers as valid? As proposed earlier, split out capabilities per M2M range with each it's own qualifiers (cost, reachability, etc) where 'unknown' could be returned as the markets worldwide develop?

Small update: the [2016 report](https://www.acm.nl/nl/publicaties/publicatie/17346/Monitor-Nummeruitgifte-2016/) from the Dutch ACM has new statistics, the amount of issued M2M numbers has gone up to 16.2 million (in a country with [17.1 million](https://www.cbs.nl/en-gb/visualisaties/population-counter) citizens).

**Comment by jbwanted** [comment]
I am writing on behalf of onoff App. We are in a process of including Netherlands numbers to our application for calling and for sending/receiving SMS-es. Due to the shortage of the 06 mobile number range, the Dutch regulator does not longer allow Apps to use the 06 numbers, but Apps are obliged to use the number range that was introduced for M2M, namely the 0970. This means that the 0970 is not limited to Machine-to-Machine communications, but used also for Human-to-Human. In fact 06 numbers can only be used with a SIM card. All calling application that want to use a mobile number range have to use 0970 or will be forced to stop by the regulator. Unfortunately regulator did not allow us to use real mobile numbers with 06 mask and threatened with a big fine, so regulator offered us M2M numbers for this purpose. 

We fully understand the policy of Libphonenumber not to support M2M number ranges, but in the Netherlands this 0970 range is the only M2M number range that is legally allowed for human-to-human calling applications.

In case of any doubts, please contact the regulator ACM

Our problem, that we use libphonenumber for number validation, as far as I know many other popular internet calling services use the same libphonenumber. So we would like kindly ask libphonenumber to add supporting of M2M numbers or at least add as a mobile range 0970050 (our numbers are from this range). There could be a case for us to do a internal workaround, but in this case our customers still would be limited with using their Netherland number in other applications.

Quoted https://www.government.nl/documents/annual-reports/2016/02/16/decree-change-of-telephone-and-isdn-services-numbering-plan-for-m2m-applications below:

The M2M services can be subdivided into four categories of applications, namely ‘Machine-to-Machine’ (M2M), ‘Machine-to-Human’, ‘Human-to-Machine’ and ‘Human-to-Human’.

**Human-to-Human**
Human-to-Human applications involve voice connections between people. In such communication, the number is usually not called automatically. In some cases, however, the number is normally called automatically. An example is a breakdown telephone in a car which can be used to call the assistance service in the event of a breakdown. The connection to the assistance service is started automatically by pressing the button. The function of this breakdown telephone is primarily to
contact the assistance service. As a rule, the device will not be called itself. For that reason, no user-friendly number is required for that device. If in the case of such an application it is possible for the assistance service to return the call to the breakdown telephone, an employee of the assistance service will initiate the communication with the breakdown telephone. Normally, however, this will be done from an automated environment. The number of the breakdown telephone will not be keyed in and therefore the device does not require a user-friendly number.

**Introduction of the 0970 range**
There are also M2M applications which use the traditional routing of traffic between different mobile and landline networks based on telephone numbers. These require cross-network numbers. In
addition, the use of cross-network numbers can sometimes be desirable to facilitate transfers to another provider, for example if the numbers are used not only for the routing but also for administrative purposes.
No distinction is made between mobile communication and communication over landline networks.
The normal legal obligations, such as number portability, apply to the 0970 numbers.

**Here is the snapshot from the latest number plan list from BICS regarding M2M numbers allocation:**
Destinations	Country Code	Area Code	Prefix Status	ValidFrom	ValidTo
Netherlands Mobile Tele2	31	9708	No Change	15-mai-2016	
Netherlands Mobile Vodafone	31	97019	No Change	15-juuni-2017	
Netherlands Mobile iBasis	31	97018	No Change	15-apr-2017	
Netherlands Mobile iBasis	31	97017	No Change	15-apr-2017	
Netherlands Mobile iBasis	31	97016	No Change	15-apr-2017	
Netherlands Mobile	31	9701500	No Change	15-sept-2016	
Netherlands Mobile Vodafone	31	970146	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970145	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970144	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970143	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970142	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970141	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970140	No Change	15-nov-2016	
Netherlands Mobile iBasis	31	97013	No Change	15-okt-2015	
Netherlands Mobile iBasis	31	97012	No Change	15-okt-2015	
Netherlands Mobile iBasis	31	97011	No Change	15-okt-2015	
Netherlands Mobile Tismi	31	970102010	No Change	15-veebr-2016	
Netherlands Mobile	31	970100	No Change	01-jaan-2015	
Netherlands Mobile	31	970094	No Change	15-mai-2014	
Netherlands Mobile	31	970090	No Change	15-mai-2014	
Netherlands Mobile	31	97006	No Change	15-märts-2014	
Netherlands Mobile Vodafone	31	970059	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970058	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970057	No Change	15-nov-2016	
Netherlands Mobile Vodafone	31	970056	No Change	01-juuni-2016	
Netherlands Mobile Vodafone	31	970055	No Change	01-juuni-2016	
Netherlands Mobile T-Mobile	31	970054	No Change	01-juuni-2016	
Netherlands Mobile T-Mobile	31	970053	No Change	01-juuni-2016	
Netherlands Mobile T-Mobile	31	970052	No Change	01-juuni-2016	
Netherlands Mobile T-Mobile	31	970051	No Change	01-juuni-2016	
Netherlands Mobile T-Mobile	31	970050	No Change	01-juuni-2016	
Netherlands Mobile	31	97005	No Change	01-juuni-2016	
Netherlands Mobile Vodafone	31	970049	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970048	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970047	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970046	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970045	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970044	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970043	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970042	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970041	No Change	01-mai-2014	
Netherlands Mobile Vodafone	31	970040	No Change	01-mai-2014	
Netherlands Mobile iBasis	31	97003	No Change	15-jaan-2014	
Netherlands Mobile iBasis	31	97002	No Change	15-jaan-2014	
Netherlands Mobile Vodafone	31	97001	No Change	15-juuni-2014	
Netherlands Mobile	31	970000	No Change	01-jaan-2015

**Comment by ameenross** [comment]
I've been trying to argue that [since 2015](https://github.com/googlei18n/libphonenumber/issues/332#issuecomment-96966593), albeit without quoting the ACM doc, as I presumed that to be unnecessary, due to the fact that it was already shared. And I expect the developers to make an informed decision and thus read all the documentation available to them...

**Comment by jbogdanov** [comment]
What writes Nexmo regarding supporting Netherlands mobile numbers
https://help.nexmo.com/hc/en-us/articles/204017903-Netherlands-Direct-route-

**Restrictions to +316*** virtual numbers (inbound SMS)**

The ACM (Consumer and Markets regulator) has decided that 06xxx numbers in the Netherlands must be attached to a mobile device. They do not consider tablets to be mobile devices either. They expect A2P type of use cases to be using the 097xx number range which was created to M2M. This range is not commercially ready due to poor oversight and lack of co-operation between the Carriers.

Nexmo has offered 06 numbers previously, however there is a risk that the service could be discontinued for those with numbers that are NOT attached to Mobile devices. There is also a risk that the regulator will ask our Carrier to stop offering the service for both Voice and SMS.

**Comment by daghoidahl** [comment]
At Dignio, our product manages pill dispensers that communicate via SMS. We would like to use libphonenumber to validate these phone numbers, and we can't, since they are in the Norwegian M2M range. So we would like some way of validating M2M numbers. We would be fine with a separate API call for this.

As for official docs about Norwegian M2M, there's something here: https://www.nkom.no/marked/nummerforvaltning/e.164-nummerplan/8-og-12-sifrede-nummerserier (translates pretty well [with Google translate](https://translate.google.com/translate?sl=no&tl=en&js=y&prev=_t&hl=no&ie=UTF-8&u=https%3A%2F%2Fwww.nkom.no%2Fmarked%2Fnummerforvaltning%2Fe.164-nummerplan%2F8-og-12-sifrede-nummerserier&edit-text=&act=url)). The document "Specifications regarding the M2M number series.pdf" looks like it contains some of the info you're looking for.

**Comment by dylmye** [comment]
(@daghoidahl's link is available in english here: https://eng.nkom.no/market/numbering/e.164-numbering-plan/8-digit-and-12-digit-number-series)

**Comment by dylmye** [comment]
There are over 200 members of the [oneM2M](http://www.onem2m.org/) standardisation initiative. oneM2M Release 2 technical specification is available [here](http://www.onem2m.org/technical/published-documents)

**Comment by andyst** [comment]
We are migrating from Github's issue tracking system to Google's Issue Tracker. Background:
https://groups.google.com/forum/#!topic/libphonenumber-discuss/-M3YjdkkOiA

Please continue the discussion of enabling M2M support in libphonenumber at:
https://issuetracker.google.com/issues/74493346

The new issue in the Google Issue Tracker summarizes the current situation, the challenges, which countries have which M2M ranges, and authoritative evidence.
